### PR TITLE
OOP Milestone game logic

### PR DESCRIPTION
**Pull request by marcelomaidden**
## OOP Milestone game logic
At this pull request I created the game logic as follows:
- Create class Game
- Create class Player
- Interaction with the user on main.rb

### CODE REVIEW

File: bin/game.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/AbcSize
+
+class Game
+  attr_accessor :on
```
Comment (kobiyoyo) [inline bin/game.rb]: - [x] Be aware that by using `attr_accessors` allows writing that property from outside the class. That variable can be set to anything.Please consider using attr_readers/attr_writers.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: bin/game.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    return player if @board[0][0] == player.alias && @board[1][1] == player.alias && @board[2][2] == player.alias
+
+    return player if @board[0][2] == player.alias && @board[1][1] == player.alias && @board[2][0] == player.alias
+
+    return player if @board[0][0] == player.alias && @board[1][0] == player.alias && @board[2][0] == player.alias
+
+    return player if @board[0][1] == player.alias && @board[1][1] == player.alias && @board[2][1] == player.alias
+
+    return player if @board[0][2] == player.alias && @board[1][2] == player.alias && @board[2][2] == player.alias
+  end
```
Comment (kobiyoyo) [inline bin/game.rb]: - [ ] DRY rule should be respected. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
Check this out
````ruby

  # better
  def check_win(player_symbol)
    # for rows
    all_equal(board[0..2], player_symbol) ||
    all_equal(board[3..5], player_symbol) ||
    all_equal(board[6..8], player_symbol)

    # and analogical as above for columns and diagonals
  end

  def all_equal(subarray, player_symbol)
    return true if subarray.uniq.size == 1 && subarray.uniq == player_symbol
  end
````

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [x] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [x] The application does inform the player about invalid input and asks for their move again
- [x] Let's use require_relative in the app because it's better than using require.
- [x] Let's implement validation for inputs, I could input integers as my name.
- [x] It will be nice if you could include  instruction on how to input columns and rows on the board(I wasn't able to properly test the interface)
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [ ] The application does inform the player about invalid input and asks for their move again
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by jubaan — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## :wave: :wave: Hello @marcelomaidden  . Thanks for submitting you project!

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] The game doesn't ends when the game reach a **draw** status
image

Most of the next issues happened after selecting `yes` in the `Do you wanna play again? [yes] [no]` question.

- [x] The assignation of the `X` and `0` is switched when the game starts, Player who was assigned the `X`, becomes the `0`.

- [x] Player is not switching correctly, it assigns consecutive turn to the same player
image
image

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

**Review by descholar-ceo — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
## :clap: :clap: :clap: Congratulations @marcelomaidden, your project is approved :tada: :tada: :tada: 
You have implemented all requested changes by previous TSEs, I am marking this project as complete, it doesn't need more changes, you can go ahead and merge all of the changes on base branch,

#### [Optional] Disabling rubocop lint rules is a bad practice, next time try hard to research and fix linters errors instead of disabling rules.

Don't forget to fill the mandatory project completion form, and see you to next project,

Happy coding :computer: :computer: :computer: 

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Pull request merged**